api/client/workflows/requestwatcher.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workflows
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/types"
+
+	"github.com/gravitational/trace"
+)
+
+// RequestWatcher is a stream watcher for Teleport access requests.
+type RequestWatcher interface {
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#interfaces interfaces belong on the calling side
this code should expose the struct that implements it only